keras_cv/models/backbones/efficientnet_v2/efficientnet_v2_backbone.py

```diff
@@ -0,0 +1,556 @@
+
+
+def round_repeats(repeats, depth_coefficient):
+    """Round number of repeats based on depth multiplier."""
+    return int(math.ceil(depth_coefficient * repeats))
+
+
+def get_block_conv(args, activation, survival_probability, name):
+    # Determine which conv type to use:
+    if args["conv_type"] == "mb_conv":
+        return MBConvBlock(
+            input_filters=args["input_filters"],
+            output_filters=args["output_filters"],
+            expand_ratio=args["expand_ratio"],
+            kernel_size=args["kernel_size"],
+            strides=args["strides"],
+            se_ratio=args["se_ratio"],
+            activation=activation,
+            bn_momentum=0.9,
+            survival_probability=survival_probability,
+            name=name,
+        )
+    elif args["conv_type"] == "fused_mb_conv":
+        return FusedMBConvBlock(
+            input_filters=args["input_filters"],
+            output_filters=args["output_filters"],
+            expand_ratio=args["expand_ratio"],
+            kernel_size=args["kernel_size"],
+            strides=args["strides"],
+            se_ratio=args["se_ratio"],
+            activation=activation,
+            bn_momentum=0.9,
+            survival_probability=survival_probability,
+            name=name,
+        )
+    raise ValueError(
+        "Expected `block_args['conv_type']` to be "
+        "one of 'mb_conv', 'fused_mb_conv', but got "
+        f"`block_args['conv_type']={args['conv_type']}`"
+    )
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class EfficientNetV2Backbone(Backbone):
+    """Instantiates the EfficientNetV2 architecture.
+
+    Args:
+        include_rescaling: bool, whether to rescale the inputs. If set
+            to `True`, inputs will be passed through a `Rescaling(1/255.0)`
+            layer.
+        width_coefficient: float, scaling coefficient for network width.
+        depth_coefficient: float, scaling coefficient for network depth.
+        default_size: integer, default input image size.
+        dropout_rate: float, dropout rate before final classifier layer.
+        drop_connect_rate: float, dropout rate at skip connections.
+        depth_divisor: integer, a unit of network width.
+        min_depth: integer, minimum number of filters.
+        activation: activation function.
+        blocks_args: list of dicts, parameters to construct block modules.
+        model_name: string, model name.
+        input_shape: optional shape tuple, defaults to (None, None, 3).
+        input_tensor: optional Keras tensor (i.e. output of `layers.Input()`)
+            to use as image input for the model.
+    """
+
+    def __init__(
+        self,
+        *,
+        include_rescaling,
+        width_coefficient,
+        depth_coefficient,
+        default_size,
+        dropout_rate=0.2,
+        drop_connect_rate=0.2,
+        depth_divisor=8,
+        min_depth=8,
+        activation="swish",
+        blocks_args="default",
```

Comment:
A couple weird things here:
* Passing configs is an antipattern in Keras. We need to find a way around this
* This is named `blocks_args` here and `block_args` above
* The default value is `"default"`, which actually throws in the helper method.
* There is no description of legal values in the docstring.

Reply:
I would recommend the pattern we developed in MobileNet instead where the block args are passed as parallel lists. See #1641

Reply:
> This is named blocks_args here and block_args above
> the default value is "default", which actually throws in the helper method.
> There is no description of legal values in the docstring.

Good points; these are definitely not correct - I'll update.

> Passing configs is an antipattern in Keras. We need to find a way around this... I would recommend the pattern we developed in MobileNet instead where the block args are passed as parallel lists. See https://github.com/keras-team/keras-cv/pull/1641

I'm not super sold on that - especially given that block types here may not even match!  In this case the configs are identical, but you can certainly imagine a scenario where theres an arg that  I.e. MbConvBlock vs FusedMBConvBlock require different arguments.

I feel we're still passing a config if its parallel lists... its just split up and harder to pick out which values go with which layers.  

Thoughts?  Personally I definitely prefer a list of dictionaries, but I suppose there is added noise comes in keys with this approach.

Reply:
the configs also render very ugly for long lists of ints: the presets in https://github.com/keras-team/keras-cv/pull/1641

Perhaps fixable with some more `black` configuration.

Reply:
Here's why I don't think this is a productive direction @LukeWood:
* You have a `block_args` param that is 100% undocumented. What fields need to be populated? What type of values are expected? This is a bad user experience.
* The most obvious way to document the params is to create an official "config" using `dataclass` or a proto
* You have just created a nested config system, which will be vetoed by @fchollet 

I do agree that the parallel lists in #1641 aren't perfect but each block param is well documented and it's easy to trace how each param evolves across the block. What we have here doesn't look like Keras code or other backbones. If we encounter a more complex case we can have a bigger design discussion, but I haven't seen one so far.

Reply:
Yep yep - sorry I mean't to comment back on this after our 1:1 discussion but I  couldnt find the thread.  Definitely updating to match the others